rasa/core/training/story_conflict.py

```diff
@@ -242,6 +255,51 @@ def _find_conflicting_states(
     }
 
 
+def _unlearnable_action(event: Event) -> bool:
+    """Identifies if the action cannot be learned by policies that use story data.
+
+    Args:
+        event: An event to be checked.
+
+    Returns:
+        `True` if the event can be learned, `False` otherwise.
+    """
+    return (
+        isinstance(event, ActionExecuted)
+        and event.action_name == ACTION_UNLIKELY_INTENT_NAME
+    )
+
+
+def _find_unlearnable_actions(
+    trackers: List[TrackerWithCachedStates],
+    domain: Domain,
+    max_history: Optional[int],
+    tokenizer: Optional[Tokenizer],
+) -> Dict[int, Optional[List[Text]]]:
+    """Identifies all states that contain actions that cannot be learned.
+
+    Args:
+        trackers: Trackers that contain the states.
+        domain: The domain object.
+        max_history: Number of turns to take into account for the state descriptions.
+        tokenizer: A tokenizer to tokenize the user messages.
+
+    Returns:
+        A dictionary mapping state-hashes to a list of action-hashes
+        that cannot be learned.
+    """
+    # Create a 'state -> list of actions' dict, where the state is
+    # represented by its hash
+    state_action_mapping = defaultdict(list)
+    for element in _sliced_states_iterator(trackers, domain, max_history, tokenizer):
```

Comment:
It works but is a bit inefficient to iterate over sliced states when we only check for action names.

Reply:
@alwx

Reply:
Updated it.

Reply:
@alwx Sorry, I wasn't clear (or GH doesn't show me the change?). It still looks like we iterate thrice over `_sliced_states_iterator`, which can be slow. Can't we do this inside `_find_conflicting_states` where the same iteration happens and have `_find_conflicting_states` return two dictionaries?